src/arroyo_reduction/operator.py

```diff
@@ -58,8 +61,26 @@ async def dispatch(self, message: RawFrameEvent) -> LatentSpaceEvent:
                 dimred_model = self.redis_model_store.get_dimred_model()
                 
                 if not autoencoder_model or not dimred_model:
+                    # NEW: Send flush only once when entering offline mode
+                    if not self._flush_sent:
+                        flush_event = LatentSpaceEvent(
+                            tiled_url="FLUSH_SIGNAL",
```

Comment:
I'm a bit confused in this section. Could you please walk me through the `flush` logic?